File: recipe-api/src/main/java/com/example/recipe_api/Product/Entities/Product.java
package com.example.recipe_api.Product.Entities;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.List;

import com.example.recipe_api.Core.Enums.Flag;
import com.example.recipe_api.Product.Enums.ProductCategory;
import com.example.recipe_api.Product.Enums.Readiness;

@Entity
@Table(name = "products")
@Data @Builder @NoArgsConstructor @AllArgsConstructor
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    private String name;
    
    @ElementCollection
    @CollectionTable(name = "product_photos", joinColumns = @JoinColumn(name = "product_id"))
    @Column(name = "photo_url")
    private List<String> photos;

    private double calories;
    private double proteins;
    private double fats;
    private double carbs;
    private String ingredients;

    private ProductCategory category;
    private Readiness readiness;

    @ElementCollection
    @CollectionTable(name = "product_flags", joinColumns = @JoinColumn(name = "product_id"))
    @Column(name = "flag")
    private List<Flag> flags;

    private LocalDateTime creationDate;
    private LocalDateTime editDate;
}

File: recipe-web/src/app/product/models/product.ts
import { Readiness } from "./readiness"
import { ProductCategory } from "./product-category"
import { Flag } from "../../core/models/flags"

export interface Product {
    id: string;
    name: string,
    photos?: (File | string)[],
    calories: number,
    proteins: number,
    fats: number,
    carbs: number,
    ingredients?: string | null,
    category: ProductCategory,
    readiness: Readiness,
    flags?: Flag[],
    creationDate?: Date,
    editDate?: Date,
}

export type ProductDraft = Omit<Product, 'id' | 'creationDate' | 'editDate'>
